src/measure_theory/measurable_space.lean

```diff
@@ -279,12 +279,8 @@ instance : measurable_space ℕ := ⊤
 instance : measurable_space ℤ := ⊤
 instance : measurable_space ℚ := ⊤
 
-instance : measurable_singleton_class empty := ⟨λ _, trivial⟩
-instance : measurable_singleton_class punit := ⟨λ _, trivial⟩
-instance : measurable_singleton_class bool := ⟨λ _, trivial⟩
-instance : measurable_singleton_class ℕ := ⟨λ _, trivial⟩
-instance : measurable_singleton_class ℤ := ⟨λ _, trivial⟩
-instance : measurable_singleton_class ℚ := ⟨λ _, trivial⟩
+instance top.measurable_singleton_class {α : Type*} : @measurable_singleton_class α ⊤ :=
```

Comment:
is this evil?

Reply:
IMHO, we should have a typeclass similar to `discrete_topology`.

Reply:
that's probably a nice idea; it's a shame we can't reuse `discrete_topology` as that is instead the bottom topology. we do end up having a bunch of typeclasses like this (`filter.ne_bot`, `discrete_topology`, `ne_zero`, this proposed one)... I wonder if we should just make a general version of `ne/is_(zero/bot/top)`

Reply:
the one issue I can think of is that the category theory library already has `category_theory.limits.is_zero`, which is genuinely different and can't really be unified

Reply:
I am also not convinced by this change. Instead, you could register a `measurable_singleton_class (fin m)` in your application file. Otherwise, LGTM.

Reply:
for now, I'll undo this change and add the specific instance. I do think we should discuss these future typeclasses on Zulip, though, so i've created a conversation [there](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/.28is.2Fne.29_.28zero.2Fbot.2Ftop.29).